plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useTinyEditor } from '../hooks/useTinyEditor';
+import MagicIcon from '../../assets/images/icons/magic.svg';
+
+type WooApiMessage = {
+	role: string;
+	content: string;
+};
+
+type WooApiResponse = {
+	generated_text: string;
+	previous_messages: WooApiMessage[];
+};
+
+const generatingContentPhrases = [
+	__( 'Please wait…', 'woocommerce' ),
+	__( 'Just a little longer…', 'woocommerce' ),
+	__( 'Almost there…', 'woocommerce' ),
+];
+
+const getGeneratingContentPhrase = () =>
+	generatingContentPhrases[
+		Math.floor( Math.random() * generatingContentPhrases.length )
+	];
+
+const DESCRIPTION_MAX_LENGTH = 300;
+
+export function WriteItForMeButtonContainer() {
+	const [ fetching, setFetching ] = useState( false );
+	const generatingContentPhraseInterval = useRef< number >();
+	const titleEl = useRef< HTMLInputElement >(
+		document.querySelector( '#title' )
+	);
+	const [ productTitle, setProductTitle ] = useState< string >(
+		titleEl.current?.value || ''
+	);
+	const tinyEditor = useTinyEditor();
+
+	useEffect( () => {
+		titleEl.current?.addEventListener( 'keyup', ( e ) =>
+			setProductTitle( ( e.target as HTMLInputElement ).value || '' )
+		);
+	}, [ titleEl ] );
+
+	useEffect( () => {
+		if ( fetching ) {
+			tinyEditor.setContent( getGeneratingContentPhrase() );
+			generatingContentPhraseInterval.current = setInterval(
+				() => tinyEditor.setContent( getGeneratingContentPhrase() ),
+				2000
+			);
+		} else {
+			clearInterval( generatingContentPhraseInterval.current );
+		}
+	}, [ fetching ] );
+
+	const buildPrompt = () => {
+		const instructions = [
+			`Write a product description with the following product title: "${ productTitle }."`,
+			'Use a 9th grade reading level.',
+			`Make the description ${ DESCRIPTION_MAX_LENGTH } words or less.`,
+			'Structure the description using standard HTML paragraph, strong and list tags.',
+			'Do not include a heading at the very top of the description.',
+		];
+
+		return instructions.join( '\n' );
+	};
+
+	return (
+		<button
+			className="button wp-media-button woo-ai-write-it-for-me-btn"
+			type="button"
+			disabled={ fetching || ! productTitle }
+			title={
+				productTitle && productTitle.length > 3
+					? undefined
+					: __(
+							'Please create a product title before generating a description.',
+							'woocommerce'
+					  )
+			}
+			onClick={ async () => {
```

Comment:
I think it would make the file more readable if the onClick() handler was moved to a named function.

Reply:
Sounds good 👍🏻